remote-api/src/main/java/org/alfresco/rest/api/impl/mapper/rules/RestRuleModelMapper.java

```diff
@@ -0,0 +1,147 @@
+import org.alfresco.rest.api.model.rules.Rule;
+import org.alfresco.rest.api.model.rules.RuleTrigger;
+import org.alfresco.service.Experimental;
+import org.alfresco.service.cmr.action.ActionCondition;
+import org.alfresco.service.cmr.action.CompositeAction;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.util.GUID;
+import org.apache.commons.collections.CollectionUtils;
+
+@Experimental
+public class RestRuleModelMapper implements RestModelMapper<Rule, org.alfresco.service.cmr.rule.Rule>
+{
+    private final RestModelMapper<CompositeCondition, ActionCondition> compositeConditionMapper;
+    private final RestModelMapper<Action, org.alfresco.service.cmr.action.Action> actionMapper;
+    private final Nodes nodes;
+
+    public RestRuleModelMapper(
+            RestModelMapper<CompositeCondition, ActionCondition> compositeConditionMapper,
+            RestModelMapper<Action, org.alfresco.service.cmr.action.Action> actionMapper,
+            Nodes nodes)
+    {
+        this.compositeConditionMapper = compositeConditionMapper;
+        this.actionMapper = actionMapper;
+        this.nodes = nodes;
+    }
+
+    /**
+     * Converts service POJO rule to REST model rule.
+     *
+     * @param serviceRule - {@link org.alfresco.service.cmr.rule.Rule} service POJO
+     * @return {@link Rule} REST model
+     */
+    @Override
+    public Rule toRestModel(org.alfresco.service.cmr.rule.Rule serviceRule)
+    {
+        if (serviceRule == null)
+        {
+            return null;
+        }
+
+        final Rule.Builder builder = Rule.builder()
+                .name(serviceRule.getTitle())
+                .description(serviceRule.getDescription())
+                .isEnabled(!serviceRule.getRuleDisabled())
+                .isInheritable(serviceRule.isAppliedToChildren())
+                .isAsynchronous(serviceRule.getExecuteAsynchronously());
+
+        if (serviceRule.getNodeRef() != null)
+        {
+            builder.id(serviceRule.getNodeRef().getId());
+        }
+        if (CollectionUtils.isNotEmpty(serviceRule.getRuleTypes()))
+        {
+            builder.triggers(serviceRule.getRuleTypes().stream().map(RuleTrigger::of).collect(Collectors.toList()));
+        }
+        if (serviceRule.getAction() != null)
+        {
+            builder.conditions(compositeConditionMapper.toRestModel(serviceRule.getAction().getActionConditions()));
+            if (serviceRule.getAction().getCompensatingAction() != null &&
+                    serviceRule.getAction().getCompensatingAction().getParameterValue(ScriptActionExecuter.PARAM_SCRIPTREF) != null)
+            {
+                builder.errorScript(
+                        serviceRule.getAction().getCompensatingAction().getParameterValue(ScriptActionExecuter.PARAM_SCRIPTREF).toString());
+            }
+            if (serviceRule.getAction() instanceof CompositeAction && ((CompositeAction) serviceRule.getAction()).getActions() != null)
```

Comment:
Is it possible that this isn't an instance of CompositeAction?  Maybe we should be logging something?

Reply:
In our current setup I don't think we can expect anything else than `CompositeAction`
I'll add logging and think we should log a ticket to do better logging.